docs/handlers/overview.md

```diff
@@ -4,8 +4,160 @@ A handler is what makes it possible to collect and render documentation for a pa
 
 ## Available handlers
 
-- [Python](python.md)
 - <a class="external" href="https://mkdocstrings.github.io/crystal/">Crystal</a>
+- <a class="external" href="https://mkdocstrings.github.io/python-legacy/">Python (Legacy)</a>
+- <a class="external" href="https://mkdocstrings.github.io/python/">Python (Experimental)</a>
+
+## About the Python handlers
+
+Since version 0.18, a new, experimental Python handler is available.
+It is based on [Griffe](https://github.com/mkdocstrings/griffe),
+which is an improved version of [pytkdocs](https://github.com/mkdocstrings/pytkdocs).
+
+Note that the experimental handler does not yet offer the same features as the legacy one.
+If you are making extensive use of the current (legacy) Python handler selection and rendering options,
+you might want to wait a bit before trying the experimental handler.
+It is also not completely ready to handle dynamically built objects,
+like classes built with a call to `type(...)`.
+For most other cases, the experimental handler will work just fine.
+
+If you want to keep using the legacy handler as long as possible,
+you can already specify the `python-legacy` extra when depending on *mkdocstrings*:
+
+```toml
+# PEP 621 dependencies declaration
+# adapt to your dependencies manager
+[project]
+dependencies = [
+    "mkdocstrings[python-legacy]>=0.18",
+]
+```
+
+The legacy handler will continue to "work" for many releases,
+as long as the new handler does not cover all previous use-cases.
+
+Using the legacy handler will emit a `UserWarning` stating that users
+should specify the `python-legacy` extra when depending on *mkdocstrings*.
+The warning will be emitted even if you do specify the extra, as we have
+no way to detect it.
+
+Warnings can be globally ignored by setting the
+[`PYTHONWARNINGS` environment variable](https://docs.python.org/3/library/warnings.html#describing-warning-filters):
+
+```bash
+PYTHONWARNINGS=ignore::UserWarning:mkdocstrings.handlers.python
+```
+
+### Migrate to the experimental Python handler
+
+To use the new, experimental Python handler,
+you must specify the `python` extra when depending on *mkdocstrings*:
+
+```toml
+# PEP 621 dependencies declaration
+# adapt to your dependencies manager
+[project]
+dependencies = [
+    "mkdocstrings[python]>=0.18",
```

Comment:
As said elsewhere, why not just suggest depending on `mkdocstrings-python`?
I think people are more likely to need to choose a version of `mkdocstrings-python` and ensure that the version of *that* is fresh enough, and then `mkdocstrings` version will be ensured fresh enough as a dependency of `mkdocstrings-python`.

Reply:
And as answered there, I like the idea of promoting some handlers as the officially supported ones.
The opposite is true: using the extra would ensure the handler's version is fresh enough and up to date with mkdocstrings' version.

I don't know. Maybe extras are indeed too far fetched. I just think there nice to have.

Reply:
I've changed the wording a bit to reflect that extras are not actually mandatory and that users can depend on the handlers directly.